src/databases.cpp

```diff
@@ -246,28 +262,55 @@ void SerializePairDistanceKVector(const Catalog &catalog, float minDistance, flo
     assert(buffer - bufferStart == SerializeLengthPairDistanceKVector(pairs.size(), numBins));
 }
 
+TetraDatabase::TetraDatabase(const unsigned char *buffer) : buffer_(buffer) {
+    maxAngle_ = *(float *)buffer;
+    buffer += sizeof(float);
+    catalogSize_ = *(int *)buffer;
```

Comment:
Both during serialization and deserialization, do not ever cast the unsigned char array to int*, short*, long*, etc because they are not guaranteed to have the same size across systems. Always explicitly use int16_t, int32_t, etc. (The exception for now is float*, which right now we do directly cast to, even though we probably shouldn't).